### ISSUE TITLE
implement __str__ for Schedular classes

### ISSUE THREAD

**Issue by jasam-sheja** [pytorch/pytorch#168967]
### 🚀 The feature, motivation and pitch

I'm logging my setup to a txt file, including model/optimizer/scheduler, and would like the string representation of schedulers to be meaningful, similar to `nn.Module` and Pytorch's Optimizers.

### Additional context

logging a scheduler: 
```
<torch.optim.lr_scheduler.OneCycleLR object at 0x14b688123e30>
```
, which is not meaningful, while logging the optimizers
```
AdamW (
Parameter Group 0
    amsgrad: False
    base_momentum: 0.85
    betas: (0.95, 0.999)
    capturable: False
    decoupled_weight_decay: True
    differentiable: False
    eps: 1e-08
    foreach: None
    fused: None
    initial_lr: 4e-05
    lr: 3.9999999999999996e-05
    max_lr: 0.001
    max_momentum: 0.95
    maximize: False
    min_lr: 4e-09
    weight_decay: 0.01
)
```

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar
